### PR TITLE
Attach the generated swagger.json for deployment with classifier is the filename of swaggerDirectory

### PR DESCRIPTION
**Pull request by dantran**

### CODE REVIEW

**Comment by dantran** [comment]
for  issue  #254

**Comment by who** [comment]
If I understand this PR, it allows you to create a maven artifact from the swagger.json file ?

If this is the case, please do the following: 

1. rename `<attachSwagger>true</attachSwagger>` to `<attachSwaggerArtifact>true</attachSwaggerArtifact>`.
2. add documentation to README.md describing the change and when/how someone would use it.

**Comment by who** [comment]
Oops, accidentally close this - sorry.  Please see my last comment for action items, @dantran

**Comment by dantran** [comment]
@who  will do. Thank you for reviewing it

**Comment by dantran** [comment]
@who not sure if I squash it right

File: README.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ You can specify several `apiSource`s. Generally, one is enough.
 | `outputPath` | The path of the generated static document, not existed parent directories will be created. If you don't want to generate a static document, just don't set it. |
 | `outputFormats` | The format types of the generated swagger spec. Valid values are `json`, `yaml` or both `json,yaml`. The `json` format is default.|
 | `swaggerDirectory` | The directory of generated `swagger.json` file. If null, no `swagger.json` will be generated. |
+| `attachSwaggerArtifact` | If enable, install/deploy the generated `swagger.json` to Maven repository using `swaggerDirectory`'s name as classifier. Default is `false`. |
+| `attachSwaggerArtifact`| The directory of generated `swagger.json` file. If null, no `swagger.json` will be generated. |
```
Comment (who) [inline README.md]: Looks like this is duplicated from the previous line.

File: README.md [hunk 2]
```diff
@@ -62,6 +62,8 @@ You can specify several `apiSource`s. Generally, one is enough.
 | `outputPath` | The path of the generated static document, not existed parent directories will be created. If you don't want to generate a static document, just don't set it. |
 | `outputFormats` | The format types of the generated swagger spec. Valid values are `json`, `yaml` or both `json,yaml`. The `json` format is default.|
 | `swaggerDirectory` | The directory of generated `swagger.json` file. If null, no `swagger.json` will be generated. |
+| `attachSwaggerArtifact` | If enable, install/deploy the generated `swagger.json` to Maven repository using `swaggerDirectory`'s name as classifier. Default is `false`. |
```
Comment (who) [inline README.md]: I think this should read:

>If enabled, the generated `swagger.json` file will be attached as a maven artifact. The `swaggerDirectory`'s name will be used as an artifact classifier. Default is `false`.

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java [hunk 3]
```diff
@@ -66,6 +65,14 @@
     @Parameter
     private String swaggerDirectory;
 
+    /**
+     * <code>attachSwaggerArtifact</code> triggers plugin execution to attach the generated
+     * wagger.json to Maven session for deployment purpose.  The attached classifier
```
Comment (who) [inline src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java]: typo here

**Comment by dantran** [comment]
@who could you squash them for me if it is not too much of your time. Thanks

**Comment by who** [comment]
Merged. Thanks @dantran

**Pull request merged**